contracts/schemes/CL4RRedeemer.sol

```diff
@@ -0,0 +1,156 @@
+pragma solidity ^0.6.12;
+// SPDX-License-Identifier: GPL-3.0
+
+import "./ContinuousLocking4Reputation.sol";
+
+/**
+ * @title A scheme for continuous locking ERC20 Token for reputation
+ */
+contract CL4RRedeemer is ArcScheme {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    using RealMath for uint216;
+    using RealMath for uint256;
+    using Math for uint256;
+
+    event Redeem(uint256 indexed _lockingId, address indexed _beneficiary, uint256 _amount, uint256 _batchIndex);
+
+    struct Batch {
+        uint256 totalScore;
+        // A mapping from lockingId to its score
+        mapping(uint256=>uint) scores;
+        mapping(uint256=>uint) redeemedScores;
+    }
+
+    struct Lock {
+        uint256 amount;
+        uint256 lockingTime;
+        uint256 period;
+    }
+
+    // A mapping from lockers addresses to their locks
+    mapping(address => mapping(uint256=>Lock)) public lockers;
+    // A mapping from batch index to batch
+    mapping(uint256 => Batch) public batches;
+
+    uint256 public reputationRewardLeft; // the amount of reputation that is still left to distribute
+    uint256 public startTime; // the time (in secs since epoch) that locking can start (is enable)
```

Comment:
can these vars be read as local vars at the redeem function ?

Reply:
Yes, but that'd be wasteful to read them every call as they are constants.

Reply:
in the cost of high deployment gas

Reply:
But then it's lower cost for using the contract. It's one time higher payment for multiple times lower payments tradeoff. I can change that but I think like this is better for gas saving.